pareto/operational_water_management/operational_produced_water_optimization_model.py

```diff
@@ -1872,393 +1841,100 @@ def TerminalProductionTankLevelBalanceRule(model, p, a, t):
-                return (
-                    model.vb_y_Flow[l, l_tilde, t] + model.vb_y_Flow[l_tilde, l, t] == 1
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_S and l_tilde in model.s_CP:
-            if model.p_SCA[l, l_tilde]:
-                return (
-                    model.vb_y_Flow[l, l_tilde, t] + model.vb_y_Flow[l_tilde, l, t] == 1
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_S and l_tilde in model.s_K:
-            if model.p_SKA[l, l_tilde]:
-                return (
-                    model.vb_y_Flow[l, l_tilde, t] + model.vb_y_Flow[l_tilde, l, t] == 1
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_S and l_tilde in model.s_R:
-            if model.p_SRA[l, l_tilde]:
-                return (
-                    model.vb_y_Flow[l, l_tilde, t] + model.vb_y_Flow[l_tilde, l, t] == 1
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_S and l_tilde in model.s_O:
-            if model.p_SOA[l, l_tilde]:
-                return (
-                    model.vb_y_Flow[l, l_tilde, t] + model.vb_y_Flow[l_tilde, l, t] == 1
-                )
+                return Constraint.Skip
+
+        model.TerminalProductionTankLevelBalance = Constraint(
+            model.s_P,
+            model.s_A,
+            model.s_T,
+            rule=TerminalProductionTankLevelBalanceRule,
+            doc="Terminal production tank level balance",
+        )
+    elif model.config.production_tanks == ProdTank.equalized:
+
+        def TerminalProductionTankLevelBalanceRule(model, p, t):
+            if t == model.s_T.last():
+                if p in model.s_P:
+                    return model.v_L_ProdTank[p, t] == model.p_lambda_ProdTank[p]
+                else:
+                    return Constraint.Skip
             else:
                 return Constraint.Skip
+
+        model.TerminalProductionTankLevelBalance = Constraint(
+            model.s_P,
+            model.s_T,
+            rule=TerminalProductionTankLevelBalanceRule,
+            doc="Terminal production tank level balance",
+        )
+    else:
+        raise Exception("storage type not supported")
+
+    def ProductionPadSupplyBalanceRule(model, p, t):
+        return (
+            model.v_B_Production[p, t]
+            == sum(model.v_F_Piped[p, l, t] for l in model.s_L if model.p_LLA[p, l])
+            + sum(model.v_F_Trucked[p, l, t] for l in model.s_L if model.p_LLT[p, l])
+            + model.v_S_Production[p, t]
+        )
+
+    model.ProductionPadSupplyBalance = Constraint(
+        model.s_PP,
+        model.s_T,
+        rule=ProductionPadSupplyBalanceRule,
+        doc="Production pad supply balance",
+    )
+
+    def CompletionsPadSupplyBalanceRule(model, p, t):
+        return (
+            model.v_B_Production[p, t]
```

Comment:
```suggestion
            p_beta_Flowback[p, t]
```
The documentation shows the beta flowback parameter here. So, the documentation and the code are inconsistent here, and I'm not sure if the code is correct or the documentation is correct.

Reply:
The production tank balance also has a discrepancy between the code and the documentation.

Reply:
This has been addressed; marking resolved.